framework/test/jest/integration/utils/pg-helper/index.js

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable no-console */
+const pgpLib = require('pg-promise');
+const childProcess = require('child_process');
+const { createStorageComponent } = require('./storage');
+
+this.pgpOptions = {
+	capSQL: true,
+	promiseLib: Promise,
+	noLocking: false,
+	noWarnings: true,
+};
+const pgp = pgpLib(this.pgpOptions);
+
+class PgHelper {
+	constructor({ dbName = 'lisk_dev' }) {
+		this.dbName = dbName;
+		this.cnStr = `postgres://localhost:5432/${dbName}`;
+		if (dbName.indexOf('postgres://') === 0) {
+			this.cnStr = dbName;
+		}
+
+		this.pgp = pgp(this.cnStr);
+		this.storage = null;
+	}
+
+	_dropDB() {
+		return new Promise(resolve => {
+			console.log('Dropping database instance:', this.dbName);
+			childProcess.exec(`dropdb ${this.dbName}`, () => resolve());
+		});
+	}
+
+	_createDB() {
+		return new Promise((resolve, reject) => {
+			console.log('Creating database instance:', this.dbName);
+			childProcess.exec(`createdb ${this.dbName}`, error => {
+				if (error) {
+					return reject(error);
+				}
+				return resolve();
+			});
+		});
+	}
+
+	async bootstrap() {
+		await this._dropDB();
+		await this._createDB();
+		return this.pgp.connect();
+	}
+
+	async cleanup() {
+		await this.storage.adapter.db.$pool.end();
+		await this.pgp.$pool.end();
+	}
+
+	async createStorage(options = {}, logger) {
+		const storageOptions = {
+			database: this.dbName,
+			user: 'lisk',
+			password: 'password',
+			min: 10,
+			max: process.env.LISK_TEST_DB_MAX_CONNECTIONS || 2,
+			logFileName: `logs/devnet/lisk_${this.dbName}.log`,
+			noWarnings: true,
+			poolIdleTimeout: 30000,
+			reapIntervalMillis: 1000,
+			logEvents: ['error'],
+			...options,
+		};
+		this.storage = await createStorageComponent(storageOptions, logger);
+
+		return this.storage;
+	}
+
+	static async createAccount(db, account) {
+		const keyValueSet = Object.keys(account).map(field => {
+			return {
+				field: `"${field}"`,
+				value:
+					field === 'publicKey'
+						? `DECODE(\${a.${field}}, 'hex')`
+						: `\${a.${field}}`,
+			};
+		});
+
+		await db.query(
+			'DELETE FROM mem_accounts WHERE "publicKey" = DECODE($1, \'hex\')',
+			account.publicKey,
+		);
+
+		await db.query(
+			`INSERT INTO mem_accounts (${keyValueSet
+				.map(k => k.field)
+				.join(',')}) VALUES (${keyValueSet.map(k => k.value).join(',')})`,
+			{ a: account },
+		);
+	}
+
+	static async getAccountByPublicKey(db, publicKey) {
+		return db.one(
+			`SELECT
```

Comment:
Would it make sense to extract this query somewhere / wrap it? Looks a bit of overwhelming here.

Reply:
Again will take care of it with #4270